--- CefBrowser_N---orig.cpp	2019-07-18 21:44:27.944045400 +0200
+++ CefBrowser_N.cpp	2019-07-19 18:34:06.303464700 +0200
@@ -1478,11 +1478,12 @@
   JNI_STATIC_DEFINE_INT(env, cls, KEY_RELEASED);
   JNI_STATIC_DEFINE_INT(env, cls, KEY_TYPED);
 
-  int event_type, modifiers;
+  int event_type, modifiers, key_code;
   char16 key_char;
   if (!CallJNIMethodI_V(env, cls, key_event, "getID", &event_type) ||
       !CallJNIMethodC_V(env, cls, key_event, "getKeyChar", &key_char) ||
-      !CallJNIMethodI_V(env, cls, key_event, "getModifiersEx", &modifiers)) {
+      !CallJNIMethodI_V(env, cls, key_event, "getModifiersEx", &modifiers) ||
+      !CallJNIMethodI_V(env, cls, key_event, "getKeyCode", &key_code)) {
     return;
   }
 
@@ -1491,10 +1492,9 @@
 
 #if defined(OS_WIN)
 
-  jlong scanCode = 0;
-  GetJNIFieldLong(env, cls, key_event, "scancode", &scanCode);
-  BYTE VkCode = LOBYTE(MapVirtualKey(scanCode, MAPVK_VSC_TO_VK));
-  cef_event.native_key_code = (scanCode << 16) |  // key scan code
+  BYTE VkCode = static_cast<BYTE>(key_code);
+   //LOBYTE(MapVirtualKey(key_code, MAPVK_VSC_TO_VK));
+  cef_event.native_key_code = (key_code << 16) |  // key scan code
                               1;                  // key repeat count
 #elif defined(OS_LINUX) || defined(OS_MACOSX)
   int key_code;
